<a href="https://colab.research.google.com/github/vjbarayuga/Week3-Pandas-for-Exploratory-Data-Manipulation-CodingDojo/blob/main/AtheleteEvents.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Average Height Exercise**

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd
df = pd.read_csv('/content/drive/MyDrive/Coding_DOJO-Data_Science_Bootcamp/02_Part_Time_Data_Science/2_Pandas_for_Data_Manipulation/Dataset/athleteEventsNoPersonal.csv')
df.head()

ID   Age  Height  ...          Sport                               Event Medal
0   5  21.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN
1   5  21.0   185.0  ...  Speed Skating  Speed Skating Women's 1,000 metres   NaN
2   5  25.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN
3   5  25.0   185.0  ...  Speed Skating  Speed Skating Women's 1,000 metres   NaN
4   5  27.0   185.0  ...  Speed Skating    Speed Skating Women's 500 metres   NaN

[5 rows x 12 columns]

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40616 entries, 0 to 40615
Data columns (total 12 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   ID      40616 non-null  int64  
 1   Age     39143 non-null  float64
 2   Height  31615 non-null  float64
 3   Team    40616 non-null  object 
 4   NOC     40616 non-null  object 
 5   Games   40616 non-null  object 
 6   Year    40616 non-null  int64  
 7   Season  40616 non-null  object 
 8   City    40616 non-null  object 
 9   Sport   40616 non-null  object 
 10  Event   40616 non-null  object 
 11  Medal   5917 non-null   object 
dtypes: float64(2), int64(2), object(8)
memory usage: 3.7+ MB


In [5]:
df.describe()

ID           Age        Height          Year
count   40616.000000  39143.000000  31615.000000  40616.000000
mean    68140.888738     25.619498    175.381117   1978.653240
std     38830.405762      6.494926     10.561791     29.850461
min         5.000000     11.000000    135.000000   1896.000000
25%     34661.750000     21.000000    168.000000   1960.000000
50%     68016.500000     25.000000    175.000000   1988.000000
75%    101475.000000     28.000000    183.000000   2004.000000
max    135571.000000     74.000000    221.000000   2016.000000

In [10]:
df.dtypes

ID          int64
Age       float64
Height    float64
Team       object
NOC        object
Games      object
Year        int64
Season     object
City       object
Sport      object
Event      object
Medal      object
dtype: object

In [11]:
df.shape

(40616, 12)

1a) Dataset has 40616 rows. 
1b) Dataset has 12 columns 
1c) Missing values were found the columns: Age, Height, and Medals. Medals has a lot of NA values likely because any athlete who did not get a medal, had no input NA for this column.

1d) 3370 entries are from London.

In [ ]:
df['City'].value_counts()

1e) Youngest athlete

In [12]:
df['Age'].min()

11.0

a. How many athletes who participated in the Sport "Tug-Of-War" were from Team "Sweden" in our sample dataset?

b. How many Gold medals were awarded to athletes who were over 40 years old?

In [ ]:
filter1 = df['Sport'] == 'Tug-Of-War'
filter2 = df['Team'] == 'Sweden'
len(df.loc[filter1 & filter2, :])

In [ ]:
gold = df['Medal'] == 'Gold'
age = df['Age'] > 40
len(df.loc[gold & age, :]) 

3a) The average height of all participants with all data included was 175.38 cm.

In [13]:
df['Height'].mean()

175.3811165585956

3b) Average height

In [14]:
df.groupby('Event')['Height'].mean()

Event
Alpine Skiing Men's Combined                      178.547170
Alpine Skiing Men's Downhill                      178.211679
Alpine Skiing Men's Giant Slalom                  176.739394
Alpine Skiing Men's Slalom                        176.482143
Alpine Skiing Men's Super G                       179.850575
                                                     ...    
Wrestling Women's Flyweight, Freestyle            158.100000
Wrestling Women's Heavyweight, Freestyle          173.444444
Wrestling Women's Light-Heavyweight, Freestyle    170.000000
Wrestling Women's Lightweight, Freestyle          162.666667
Wrestling Women's Middleweight, Freestyle         164.533333
Name: Height, Length: 726, dtype: float64

The average height when each athlete was only represented once was 176.32 cm.

In [15]:
one_per_athlete = df.drop_duplicates(subset = 'ID', keep = 'first')

In [16]:
one_per_athlete['Height'].mean()

176.31818479042698